src/sentry/web/frontend/auth_login.py

```diff
@@ -129,11 +131,25 @@ def _handle_login(self, request: Request, user, organization: Optional[Organizat
         )
 
     def handle_basic_auth(self, request: Request, **kwargs):
-        can_register = self.can_register(request)
-
         op = request.POST.get("op")
         organization = kwargs.pop("organization", None)
 
+        if (
+            request.method == "GET"
+            and request.subdomain
+            and Organization.objects.filter(
+                slug=request.subdomain, status=OrganizationStatus.VISIBLE
+            ).exists()
```

Comment:
Seems like a place where we should be using the `OrganizationService.get_organization_by_slug()` so that we don't make more work for future us.